web/src/lightbox.js

```diff
@@ -241,8 +255,27 @@
     render_lightbox_list_images(payload.preview);
 
     $(
-        "#lightbox_overlay .image-preview, .image-description, .download, .lightbox-zoom-reset",
+        "#lightbox_overlay .image-preview, .image-description, .download, .lightbox-zoom-reset, .video-player",
     ).hide();
+    $(".player-container").show();
+
+    if (payload.type === "inline-video") {
+        $(".player-container").hide();
+        $(".video-player, .image-description").show();
+        const $video = $("<video>");
+        $video.attr("src", payload.source);
```

Comment:
@amanagr the CodeQL alerts for `src` and `href` here are because we need to validate that `payload.source` is not validated that it's a URL, so it could be a `javascript:` item. The server doesn't ever emit that, but it seems worth defending against the possibility, ideally through `parse_image_data` validating that all the URLs its working with are in fact URLs.

I think one option would be to move `is_valid_url` to `web/src/util.ts`, reimplement it to not use regexes (https://snyk.io/blog/secure-javascript-url-validation/), and then call that for each of the URLs that we're returning from the parsing process here.

Reply:
We also have `isURL()`, which was introduced in #26294. I don't think it will validate a `javascript:` URL, either. But [its tests](https://github.com/segmentio/is-url/blob/master/test/index.js) aren't explicit about that.

Reply:
Done.